cli/index.js

```diff
@@ -0,0 +1,178 @@
+} else {
+   allPokemonNames = JSON.parse(fs.readFileSync(POKEMON_FILE_NAME));
+}
+
+async function getAllPokemonNames() {
+   const pokemonNames = await pokemonClient.getAllPokemonNames();
+   return JSON.stringify(pokemonNames);
+}
+
+const colorOptions = {
+   type: "string",
+   text: { value: "white" },
+   bg: { value: "blue" },
+};
+
+const program = new Command();
+
+program
+   .name(chalklet.generate("Weekend To-Do", colorOptions))
+   .description(
+      chalk.bgBlueBright(
+         "Get your tasks done before the weekend! And catch some Pokemons while you're at it..."
+      )
+   )
+   .version("1.0.0");
+
+program
+   .command("add")
+   .description("Add a task to your to-do list")
+   .argument("<string>", "task")
+   .action(async (task) => {
+      addTask(task);
+   });
+
+program
+   .command("delete")
+   .description("Delete a task of your choice")
+   .action(async () => {
+      const index = await chooseTaskToDelete();
+      tasksManager.remove(index);
+      console.log(chalk.bgRedBright("Todo deleted successfully"));
+   });
+
+program
+   .command("clear")
+   .description("Clear entire to-do list")
+   .action(() => {
+      tasksManager.clear();
+      console.log(chalk.red.bgGreen("All tasks cleared"));
+   });
+
+program
+   .command("get")
+   .description("Get entire to-do list")
+   .action(() => {
+      tasksManager.items.forEach((item) => {
+         console.log(chalk.bgCyan(item));
+      });
+   });
+
+program
+   .command("sort")
+   .description("Sort the To-Do list up or down")
+   .argument("<string>", "direction")
+   .action((direction) => {
+      if (direction === "up") {
+         tasksManager.sortUp();
+      } else if (direction === "down") {
+         tasksManager.sortDown();
+      } else {
+         console.log(chalk.bgRed("Unfamiliar direction argument. Please type up or down"));
+      }
+   });
+
+program.parse();
+
+function render() {
+   let data = JSON.stringify(tasksManager.items);
+   fs.writeFileSync(TASKS_FILE_NAME, data);
+}
+
+async function chooseTaskToDelete() {
+   const answer = await inquirer.prompt({
+      name: "delete",
+      type: "list",
+      message: "Which task would you like to delete?",
+      choices: tasksManager.items,
+   });
+   return tasksManager.items.indexOf(answer.delete);
+}
+
+async function addTask(task) {
+   if (
+      task
+         .replace(/\s/g, "")
+         .split(",")
+         .every((elem) => !isNaN(elem) || allPokemonNames.includes(elem))
+   ) {
+      let pokemonIDS = task.replace(/\s/g, "").split(","); // "1, 2, 3" => [1,2,3]
+      const pokemonData = await Promise.all(pokemonIDS.map((id) => pokemonClient.getPokemon(id)));
+      pokemonData.forEach((pokemon, i) => {
+         pokemonTasksHandle(pokemon, pokemonIDS, i);
+      });
+   } else {
+      tasksManager.add(task);
+      console.log(chalk.bgGreen("New todo added successfully"));
+   }
+}
+
+function pokemonTasksHandle(pokemon, pokemonIDS, i) {
+   if (pokemon) {
```

Comment:
all of this business logic should not be part of the cli file. 
We want to separate concerns. the cli file should only handle the cli interface and call the relevant actions in our business logic layer, ItemManager in this case.

Reply:
Right, great idea thanks. Just moved them.